projects/js-packages/publicize-components/src/components/media-section/index.js

```diff
@@ -33,41 +39,69 @@ const getMediaDetails = media => {
 		mediaWidth: mediaObject.width,
 		mediaHeight: mediaObject.height,
 		mediaSourceUrl: mediaObject.source_url,
+		fullFileSize: media.media_details.filesize,
+		mime: media.mime_type,
 	};
 };
 
 const ADD_MEDIA_LABEL = __( 'Set Social Image', 'jetpack' );
 const REPLACE_MEDIA_LABEL = __( 'Replace Social Image', 'jetpack' );
 const REMOVE_MEDIA_LABEL = __( 'Remove Social Image', 'jetpack' );
-
-const ALLOWED_MEDIA_TYPES = [ 'image/jpeg', 'image/png' ];
+const FILE_TYPE_ERROR_CODE = 1;
+const FILE_SIZE_ERROR_CODE = 2;
 
 /**
  * Wrapper that handles media-related functionality.
  *
  * @returns {object} The media section.
  */
 export default function MediaSection() {
+	const [ validationError, setValidationError ] = useState( null );
 	const { attachedMedia, updateAttachedMedia } = useAttachedMedia();
+	const { enabledConnections } = useSocialMediaConnections();
+
+	const { maxImageSize, allowedMediaTypes, getValidationError } = useMediaRestrictions(
+		enabledConnections
+	);
 
-	const mediaObject = useSelect( select =>
-		select( 'core' ).getMedia( attachedMedia[ 0 ] || null, { context: 'view' } )
+	const mediaObject = useSelect(
+		select => select( 'core' ).getMedia( attachedMedia[ 0 ] || null, { context: 'view' } ),
+		[ attachedMedia[ 0 ] ]
+	);
+	const { mediaWidth, mediaHeight, mediaSourceUrl, fullFileSize, mime } = getMediaDetails(
+		mediaObject
 	);
 
-	const { mediaWidth, mediaHeight, mediaSourceUrl } = getMediaDetails( mediaObject );
+	useEffect( () => {
+		// Removes selected media if connection change results in invalid image
+		if ( ! mediaObject ) {
+			return;
+		}
+
+		const error = getValidationError( fullFileSize, mime );
+		if ( error ) {
+			setValidationError( error );
+			updateAttachedMedia( [] );
+		}
+	}, [ updateAttachedMedia, mediaObject, getValidationError, fullFileSize, mime ] );
```

Comment:
There's something about this `useEffect` the `validationError` state and the `onUpdateMedia` callback, that I think could be better, but I'm not sure what it is!

On mount we'll initialise state, so we should be able to change the `null` to the any current validation errer. Then things should only change as a result of the media changing, so that would be handled in `onUpdateMedia`. Would that mean we could remove the `useEffect`?

Reply:
I think the state can change not only on `onUpdateMedia` but on connection changes as well. This is why I introduced this useEffect hook, so we can react to that, not only when the page refreshes

Reply:
I played with this but cannot come up with a solution that works without having the useEffect hook. Maybe we can come back to this as a research spike to see if there is a better way of doing this?